### ISSUE TITLE
Carousel bug: can't click child element

### ISSUE THREAD

**Issue by TimPchelintsev** [grommet/grommet#3536]
### Expected Behavior

Thanks for a great lib! I embed Facebook video as one of Carousel children. I want to play it.

### Actual Behavior

Can't play video. Click is not propagated, checked with console.log inside onClick, no output.

### URL, screen shot, or Codepen exhibiting the issue

[![Edit Grommet v2 Template](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/grommet-v2-template-2q4tc?fontsize=14)

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version: 2.7.10
- Browser Name and version: Chrome 79.0.3945.8
- Operating System and version (desktop or mobile): macOS Catalina 10.15

**Comment by ShimiSun** [comment]
hi buddy, I don't see the video link or attachment, what am I missing?
It will be easier for us to provide a quicker solution if you can reproduce the issue on a codesandbox (here is a template https://codesandbox.io/s/m7mml8l0zj) or on one of our existing storybook examples https://storybook.grommet.io/?path=/story/all--all

**Comment by TimPchelintsev** [comment]
Thank you for quick response, I have added link to initial comment, but anyway:
[![Edit Grommet v2 Template](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/grommet-v2-template-2q4tc?fontsize=14)

**Comment by wortkotze** [comment]
any update for this ?

**Comment by simonhambly** [comment]
Looking at the code of Carousel, it's using the Stack component. The browser pointer events are being blocked by the Stack layer that holds the carousel controls (and by child layers higher in the Stack).

As a workaround you can wrap this custom theme around the Carousel

```jsx
const pointerEventsFriendlyTheme = {
  box: {
    extend: 'pointer-events: none;'
  },
  anchor: {
    extend: 'pointer-events: auto;'
  },
  button: {
    extend: 'pointer-events: auto;'
  }
}

// in the component
<ThemeContext.Extend value={pointerEventsFriendlyTheme}>
  <Carousel >
    {/* carousel children*/}
  </Carousel>
</ThemeContext.Extend>
```


This disables pointer-events on Box elements (and all children), but re-enables them specifically for Anchors or Buttons.

**Comment by simonhambly** [comment]
Word of warning on the above quick fix, if your click targets in child elements are in different places, you may get 'phantom' clicks as all elements will receive clicks, even if they are not on the visible child layer.

Ideally enabling pointer-events on child controls should only be done on the active child layer.

**Comment by dokterjay19** [comment]
I believe I am having the same issue, except it is not with a stacked component, but rather with an iframe nested within a Box nested within a Grid.

Tried the work around a multiple levels of the component, but still cannot see the issue resolved. Clicking on the iframe results in the click event being registered to one of the divs (I was able to print out the styled box classname of the div to verify).

Any ideas and/or is this a separate issue?

**Comment by ShimiSun** [comment]
@dokterjay19 sounds like a separate issue, please attach a codesandbox that reproduces the issue.

**Comment by buoyantair** [comment]
Hi I would love to take this issue...

**Comment by ShimiSun** [comment]
Thanks @buoyantair!

**Comment by IanKBovard** [comment]
Have a pr open for this but, still in discussion
https://github.com/grommet/grommet/issues/3102

**Comment by heyayush** [comment]
Any update on this?

**Comment by matteod** [comment]
Any update on thi bug?

**Comment by buoyantair** [comment]
@matteod @heyayush I am currently not working on this bug, please feel free to pick it up!!

**Comment by Xtrazyx** [comment]
Any update please or a proper work around ?

**Comment by CopyJosh** [comment]
For me, the `ThemeContext.Extend` suggestion by itself did not work. Hopefully this is quickly irrelevant after https://github.com/grommet/grommet/pull/5741 but here's what I did.

Smartly, the `Carousel` component spreads props onto the `Stack` component, so it's possible if your only interactive slide is the first slide, set `controls={false}` on the carousel, wrap your own controls, and add `interactiveChild="first"`.

This worked as expected. But then I needed the last slide to have a CTA. Unfortunately, a little design flaw: even if controls are disabled and hidden, the controls are wrapped in a `Box` that still renders as the last element. So the empty div will be targeted when `interactiveChild="last"` ... what a headache lol.

I wrapped the `Carousel` and hid that invisible last element. So I was able to hack it with the following:

```js
const StyledCarousel= styled(Carousel)`
 > * {
   &:last-child {
     display: none; /* hide the empty controls and selectors div */
   }
 }
`;

const pointerEventsFriendlyTheme = {
  box: {
    extend: 'pointer-events: none;',
  },
  anchor: {
    extend: 'pointer-events: auto;',
  },
  button: {
    extend: 'pointer-events: auto;',
  },
};
...
<ThemeContext.Extend value={pointerEventsFriendlyTheme}>
  <CarouselWithoutControlsLayer
    ...
    controls={false}
    interactiveChild="last"
  >
    {props.children}
  </CarouselWithoutControlsLayer>
</ThemeContext.Extend>
```

**Comment by kevingill55** [comment]
Yeah you shouldn't need to do that once that PR above goes in! Should be in very soon 🤝
